src/main/java/org/gradoop/benchmarks/tpgm/DiffBenchmark.java

```diff
@@ -0,0 +1,160 @@
+    OPTIONS.addRequiredOption(OPTION_QUERY_2, "queryts2", true, "Used second query timestamp [ms]");
+    OPTIONS.addOption(OPTION_VERIFICATION, "verification", false, "Verify Snapshot with join.");
+  }
+
+  /**
+   * Main program to run the benchmark. Arguments are the available options.
+   * Example: {@code /path/to/flink run -c org.gradoop.benchmark.tpgm.SnapshotBenchmark
+   * path/to/gradoop-examples.jar -i hdfs:///graph -o hdfs:///output -c results.csv
+   * -x 1287000000000 -y 1230000000000}
+   *
+   * @param args program arguments
+   * @throws Exception in case of error
+   */
+  public static void main(String[] args) throws Exception {
+    CommandLine cmd = parseArguments(args, DiffBenchmark.class.getName());
+
+    if (cmd == null) {
+      return;
+    }
+
+    // read cmd arguments
+    readBaseCMDArguments(cmd);
+    readCMDArguments(cmd);
+
+    // create gradoop config
+    ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+    TemporalGradoopConfig conf = TemporalGradoopConfig.createConfig(env);
+
+    // read graph
+    TemporalDataSource source = new TemporalCSVDataSource(INPUT_PATH, conf);
+    TemporalGraph graph = source.getTemporalGraph();
+
+    // get the diff
+    TemporalGraph diff = graph.diff(new AsOf(QUERY_FROM_1), new AsOf(QUERY_FROM_2));
+
+    // apply optional verification
+    if (VERIFICATION) {
+      diff = diff.verify();
+    }
+
+    // write graph
+    writeOrCountGraph(diff, conf);
+
+    // execute and write job statistics
+    env.execute(DiffBenchmark.class.getSimpleName() + " - P: " + env.getParallelism());
+    writeCSV(env);
+  }
+
+  /**
+   * Reads the given arguments from command line
+   *
+   * @param cmd command line
+   */
+  private static void readCMDArguments(CommandLine cmd) {
+    QUERY_FROM_1 = Long.valueOf(cmd.getOptionValue(OPTION_QUERY_1));
+    QUERY_FROM_2 = Long.valueOf(cmd.getOptionValue(OPTION_QUERY_2));
+    VERIFICATION = cmd.hasOption(OPTION_VERIFICATION);
+  }
+
+  /**
+   * Method to create and add lines to a csv-file
+   *
+   * @param env given ExecutionEnvironment
+   * @throws IOException exeption during file writing
+   */
+  private static void writeCSV(ExecutionEnvironment env) throws IOException {
+    String head = String
+      .format("%s|%s|%s|%s|%s|%s%n",
+        "Parallelism",
+        "dataset",
+        "asOf1(ms)",
+        "asOf2(ms)",
+        "verify",
+        "Runtime(s)");
+
+    String tail = String
+      .format("%s|%s|%s|%s|%s|%s%n",
+        env.getParallelism(),
+        INPUT_PATH,
+        QUERY_FROM_1,
+        QUERY_FROM_2,
+        VERIFICATION,
+        env.getLastJobExecutionResult().getNetRuntime(TimeUnit.SECONDS));
+
+    File f = new File(CSV_PATH);
+    if (f.exists() && !f.isDirectory()) {
+      FileUtils.writeStringToFile(f, tail, true);
+    } else {
+      PrintWriter writer = new PrintWriter(CSV_PATH, "UTF-8");
+      writer.print(head);
+      writer.print(tail);
+      writer.close();
```

Comment:
Same thing here, this looks odd with the 2 different ways of writing the file (also missing charset parameter of the `FileUtils` method)